File: app/src/main/java/com/example/calculator/controller/TranActivity.java
package com.example.calculator.controller;

import android.os.Bundle;
import android.widget.EditText;
import android.widget.RadioGroup;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.example.calculator.R;

import java.math.BigDecimal;

public class TranActivity extends AppCompatActivity {

    EditText lengthValue, volumeValue;
    TextView lengthResult, volumeResult;
    RadioGroup lengthValues, lengthResults, volumeValues, volumeResults;

    BigDecimal length, volume;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_tran);
        initComponents();
    }

    private void initComponents() {
        lengthValue = findViewById(R.id.length_value_text);
        volumeValue = findViewById(R.id.volume_value_text);

        lengthResult = findViewById(R.id.length_result_text);
        volumeResult = findViewById(R.id.volume_result_text);

        lengthValues = findViewById(R.id.length_values);
        lengthResults = findViewById(R.id.length_results);
        lengthValues.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(RadioGroup group, int checkedId) {
                BigDecimal input = new BigDecimal(lengthValue.getText().toString());
                switch (checkedId) {
                    case R.id.length_value_km:
                        length = input.multiply(new BigDecimal("1000"));
                        break;
                    case R.id.length_value_m:
                        length = input;
                        break;
                    case R.id.length_value_cm:
                        length = input.multiply(new BigDecimal("0.01"));
                        break;
                    case R.id.length_value_mm:
                        length = input.multiply(new BigDecimal("0.001"));
                        break;
                }
            }
        });
        lengthResults.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(RadioGroup group, int checkedId) {
                String result;
                BigDecimal temp = new BigDecimal(0);
                switch (checkedId) {
                    case R.id.length_result_km:
                        temp = length.multiply(new BigDecimal("0.001"));
                        break;
                    case R.id.length_result_m:
                        temp = length;
                        break;
                    case R.id.length_result_cm:
                        temp = length.multiply(new BigDecimal("100"));
                        break;
                    case R.id.length_result_mm:
                        temp = length.multiply(new BigDecimal("1000"));
                        break;
                }
                result = temp.toString();
                lengthResult.setText(result);
            }
        });


        volumeValues = findViewById(R.id.volume_values);
        volumeResults = findViewById(R.id.volume_results);
        volumeValues.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(RadioGroup group, int checkedId) {
                BigDecimal input = new BigDecimal(volumeValue.getText().toString());
                switch (checkedId) {
                    case R.id.volume_value_m2:
                        volume = input.multiply(new BigDecimal("1000"));
                        break;
                    case R.id.volume_value_dm2:
                        volume = input;
                        break;
                    case R.id.volume_value_cm2:
                        volume = input.multiply(new BigDecimal("0.001"));
                        break;
                }
            }
        });
        volumeResults.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(RadioGroup group, int checkedId) {
                String result;
                BigDecimal temp = new BigDecimal(0);
                switch (checkedId) {
                    case R.id.volume_result_m2:
                        temp = volume.multiply(new BigDecimal("0.001"));
                        break;
                    case R.id.volume_result_dm2:
                        temp = volume;
                        break;
                    case R.id.volume_result_cm2:
                        temp = volume.multiply(new BigDecimal("1000"));
                        break;
                }
                result = temp.toString();
                volumeResult.setText(result);
            }
        });
    }

}

File: app/src/main/java/com/example/calculator/controller/TranActivity.kt
package com.example.calculator.controller

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.calculator.R
import android.widget.RadioGroup
import android.widget.TextView
import android.widget.EditText
import java.math.BigDecimal


class TranActivity: AppCompatActivity() {
    lateinit var lengthValue: EditText
    lateinit var volumeValue:EditText
    lateinit var lengthResult: TextView
    lateinit var volumeResult:TextView
    lateinit var lengthValues: RadioGroup
    lateinit var lengthResults:RadioGroup
    lateinit var volumeValues:RadioGroup
    lateinit var volumeResults:RadioGroup
    lateinit var length: BigDecimal
    lateinit var volume: BigDecimal

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_tran)
        initComponents()
    }

    private fun initComponents() {
        lengthValue = findViewById(R.id.length_value_text)
        volumeValue = findViewById<EditText>(R.id.volume_value_text)
        lengthResult = findViewById(R.id.length_result_text)
        volumeResult = findViewById<TextView>(R.id.volume_result_text)
        lengthValues = findViewById(R.id.length_values)
        lengthResults = findViewById<RadioGroup>(R.id.length_results)
        lengthValues.setOnCheckedChangeListener(RadioGroup.OnCheckedChangeListener { group, checkedId ->
            val input = BigDecimal(lengthValue.getText().toString())
            when (checkedId) {
                R.id.length_value_km -> length = input.multiply(BigDecimal("1000"))
                R.id.length_value_m -> length = input
                R.id.length_value_cm -> length = input.multiply(BigDecimal("0.01"))
                R.id.length_value_mm -> length = input.multiply(BigDecimal("0.001"))
            }
        })
        lengthResults.setOnCheckedChangeListener(RadioGroup.OnCheckedChangeListener { group, checkedId ->
            val result: String
            var temp = BigDecimal(0)
            when (checkedId) {
                R.id.length_result_km -> temp = length.multiply(BigDecimal("0.001"))
                R.id.length_result_m -> temp = length
                R.id.length_result_cm -> temp = length.multiply(BigDecimal("100"))
                R.id.length_result_mm -> temp = length.multiply(BigDecimal("1000"))
            }
            result = temp.toString()
            lengthResult.setText(result)
        })
        volumeValues = findViewById<RadioGroup>(R.id.volume_values)
        volumeResults = findViewById<RadioGroup>(R.id.volume_results)
        volumeValues.setOnCheckedChangeListener(RadioGroup.OnCheckedChangeListener { group, checkedId ->
            val input = BigDecimal(volumeValue.getText().toString())
            when (checkedId) {
                R.id.volume_value_m2 -> volume = input.multiply(BigDecimal("1000"))
                R.id.volume_value_dm2 -> volume = input
                R.id.volume_value_cm2 -> volume = input.multiply(BigDecimal("0.001"))
            }
        })
        volumeResults.setOnCheckedChangeListener(RadioGroup.OnCheckedChangeListener { group, checkedId ->
            val result: String
            var temp = BigDecimal(0)
            when (checkedId) {
                R.id.volume_result_m2 -> temp = volume.multiply(BigDecimal("0.001"))
                R.id.volume_result_dm2 -> temp = volume
                R.id.volume_result_cm2 -> temp = volume.multiply(BigDecimal("1000"))
            }
            result = temp.toString()
            volumeResult.setText(result)
        })
    }


}